### PR TITLE
jetporch init at unstable-2023-10-16

### PR DESCRIPTION
**Pull request by heywoodlh**
## Description of changes

Initial commit of [JetPorch](https://www.jetporch.com/) version unstable-2023-10-16 [rev 6ca8a26c0514a31c359bad081ef55980dd509912](https://git.sr.ht/~mpdehaan/jetporch/tree/6ca8a26c0514a31c359bad081ef55980dd509912).

## Things done

~Plan to build on Linux later, just wanted to get the PR put together.~

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/admin/jetporch/default.nix
Comment (FliegendeWurst) [inline pkgs/tools/admin/jetporch/default.nix:17]: The source is at https://git.sr.ht/~mpdehaan/jetporch now.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/tools/admin/jetporch/default.nix
Comment (heywoodlh) [reply pkgs/tools/admin/jetporch/default.nix]: Fixed in a3b2dcbaa5c0028f086b72dc1d651f28a3c28841

File: pkgs/tools/admin/jetporch/default.nix
Comment (heywoodlh) [reply pkgs/tools/admin/jetporch/default.nix]: Fixed in a3b2dcbaa5c0028f086b72dc1d651f28a3c28841

Comment (FliegendeWurst) [reply pkgs/tools/admin/jetporch/default.nix]: The download still does not work. In fact I haven't found any versioned releases in that repo... I think it would be best to use the "unstable-YYYY-MM-DD" version scheme. See https://nixos.org/manual/nixpkgs/stable/#sec-package-naming

Comment (heywoodlh) [reply pkgs/tools/admin/jetporch/default.nix]: @FliegendeWurst can you check the amended commit now? Let me know if that works. 

(Thank you for the thoughtful review, btw)

File: pkgs/tools/admin/jetporch/default.nix
Comment (FliegendeWurst) [inline pkgs/tools/admin/jetporch/default.nix:6]:
```suggestion
{ lib,
  rustPlatform,
  fetchFromSourcehut,
  openssl,
  pkg-config,
}:
```
`...` is generally not used for package definitions in nixpkgs.

File: pkgs/tools/admin/jetporch/default.nix
Comment (FliegendeWurst) [inline pkgs/tools/admin/jetporch/default.nix:10]:
```suggestion
  version = "unstable-2023-10-16";
```
The date has to be the commit date.
(There is also a newer commit from seven hours ago, but it doesn't look relevant for our package.)

File: pkgs/tools/admin/jetporch/default.nix
Comment (FliegendeWurst) [inline pkgs/tools/admin/jetporch/default.nix:18]:
```suggestion
  buildInputs = [
    openssl
  ];
```

File: pkgs/top-level/all-packages.nix
Comment (FliegendeWurst) [inline pkgs/top-level/all-packages.nix:9555]: I think the new best practice is to have packages in pkgs/by-name.
For jetporch, the location would be `pkgs/by-name/je/jetporch/package.nix`.
That way you do not need to add the package to all-packages.nix

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Just some code style issues remaining.

Please also reorganize the commits to be "maintainers: add heywoodlh" and "jetporch: init at unstable-2023-10-16".

File: pkgs/top-level/all-packages.nix
Comment (heywoodlh) [reply pkgs/top-level/all-packages.nix:9555]: So would I put the package in `pkgs/by-name/je/jetporch/package.nix` instead of `pkgs/tools/admin/jetporch/default.nix`?

Comment (heywoodlh) [reply pkgs/top-level/all-packages.nix]: Done! 7ab4cae0c31fd3f02a42ca31aedf8996f31b8738

File: pkgs/tools/admin/jetporch/default.nix
Comment (heywoodlh) [reply pkgs/tools/admin/jetporch/default.nix]: Done! d66749979e721343e7c86693d4490a98d213b9f8

File: pkgs/tools/admin/jetporch/default.nix
Comment (heywoodlh) [reply pkgs/tools/admin/jetporch/default.nix]: Done: d66749979e721343e7c86693d4490a98d213b9f8

**Comment by heywoodlh** [comment]
@FliegendeWurst I think I've completed all your suggestions, please let me know if there is anything else I can/should do.

**Comment by FliegendeWurst** [comment]
> @FliegendeWurst I think I've completed all your suggestions, please let me know if there is anything else I can/should do.

Yes, the commits should be the other way around. Otherwise the first commit doesn't evaluate.

After that's done we just need to wait for someone to merge the PR (I don't have the required permissions)

**Comment by heywoodlh** [comment]
@FliegendeWurst order should be fixed.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by heywoodlh** [comment]
This project has been discontinued. The original page is dead: https://www.jetporch.com/

Since I can't link to the original post on JetPorch, here's a community-contributed post: https://lucaberton.medium.com/farewell-to-jetporch-automation-b820021e512f

**Pull request closed without merging**
